contribs/gmf/src/directives/print.js

```diff
@@ -365,84 +377,95 @@ gmf.PrintController = function($rootScope, $scope, $timeout, $q, $injector,
-   * @return {number} Scale of the map to print.
+   * @type {angular.$http.HttpPromise}
+   * @private
    */
-  const getScaleFn = (frameState) => {
-    // Don't compute an optimal scale if the user manualy choose a value not in
-    // the pre-defined scales. (`scaleInput` in `gmfPrintOptions`).
-    goog.asserts.assert(this.layoutInfo.scales);
-    goog.asserts.assert(this.layoutInfo.scale !== undefined);
-    if (this.layoutInfo.scale === -1 ||
-        ol.array.includes(this.layoutInfo.scales, this.layoutInfo.scale)) {
-      const mapSize = frameState.size;
-      const viewResolution = frameState.viewState.resolution;
-      this.layoutInfo.scale = this.getOptimalScale_(mapSize, viewResolution);
-    }
-    return this.layoutInfo.scale;
-  };
-
-  let getRotationFn;
-  if (this.rotateMask_) {
-    /**
-     * @return {number} rotation to apply.
-     */
-    getRotationFn = () => this.rotation;
-  }
+  this.capabilities_;
 
   /**
-   * @type {function(ol.render.Event)}
+   * @type {gmfThemes.GmfOgcServers}
+   * @private
    */
-  this.postcomposeListener_ = ngeoPrintUtils.createPrintMaskPostcompose(
-    getSizeFn, getScaleFn, getRotationFn);
+  this.ogcServers_;
 
   /**
-   * @type {angular.$http.HttpPromise}
+   * @type {Array.<gmfThemes.GmfTheme>}
    * @private
    */
-  this.capabilities_;
+  this.currentThemes_;
+};
 
+
+/**
+ * Init the controller
+ */
+gmf.PrintController.prototype.$onInit = function() {
   // Clear the capabilities if the roleId changes
-  $scope.$watch(() => gmfAuthentication.getRoleId(), () => {
+  this.$scope_.$watch(() => this.gmfAuthentication_.getRoleId(), () => {
     this.gmfPrintState_.state = gmf.PrintStateEnum.CAPABILITIES_NOT_LOADED;
     this.capabilities_ = null;
   });
 
+  this.$scope_.$watch(() => this.active, (active) => {
+    this.togglePrintPanel_(active);
+  });
+
   // Print on event.
-  $rootScope.$on('gmfStartPrint', (event, format) => {
+  this.$rootScope_.$on('gmfStartPrint', (event, format) => {
     this.print(`${format}`);
   });
 
   // Cancel print task on event.
-  $rootScope.$on('gmfCancelPrint', () => {
+  this.$rootScope_.$on('gmfCancelPrint', () => {
     this.cancel();
   });
 
+  this.gmfThemes_.getOgcServersObject().then((ogcServersObject) => {
+    this.ogcServers_ = ogcServersObject;
+  });
+
+  this.gmfThemes_.getThemesObject().then((currentThemes) => {
+    this.currentThemes_ = currentThemes;
+  });
 
   /**
-   * @type {gmfThemes.GmfOgcServers}
-   * @private
+   * @return {ol.Size} Size in dots of the map to print.
    */
-  this.ogcServers_;
-
-  gmfThemes.getOgcServersObject().then((ogcServersObject) => {
-    this.ogcServers_ = ogcServersObject;
-  });
+  const getSizeFn = () => this.paperSize_;
 
   /**
-   * @type {Array.<gmfThemes.GmfTheme>}
-   * @private
+   * @param {olx.FrameState} frameState Frame state.
+   * @return {number} Scale of the map to print.
    */
-  this.currentThemes_;
+  const getScaleFn = (frameState) => {
```

Comment:
why not a method? it seems a bit complex arrow-function

Reply:
I agree, but I'm not here tomorrow, if I don't merge now, I'll merge Monday :-/